src/core/metrics/constants.ts

```diff
@@ -12,14 +17,39 @@ export const METRICS_REPORT_TEMPLATE: MetricsReportTemplate = {
   buildVersion: `${import.meta.env.PACKAGE_VERSION}-${import.meta.env.MODE}`,
 };
 
-// ! Relevant only for map mode
-export const METRICS_WATCH_LIST = {
-  // _done_userResourceAtom: null, // should be done before init
-  _done_currentEventResource: null,
-  _done_layersGlobalResource: null,
-  _done_analyticsResource: null,
+const APPEVENT_TO_FEATURE = {
+  [AppFeature.CURRENT_EVENT]: AppFeature.CURRENT_EVENT, // can be error
+  _done_layersGlobalResource: AppFeature.LAYERS_IN_AREA, // list of layers
+  [AppFeature.ANALYTICS_PANEL]: [AppFeature.ANALYTICS_PANEL, AppFeature.CURRENT_EVENT], // not in EPIG, depends on CURRENT_EVENT
   // _done_areaLayersDetailsResourceAtom: null, // can be disabled in url
-  _done_layersInAreaAndEventLayerResource: null,
-  // _done_eventListResource: null,
-  [EVENT_MAP_IDLE]: null,
+  _done_layersInAreaAndEventLayerResource: [
+    AppFeature.LAYERS_IN_AREA,
+    AppFeature.CURRENT_EVENT,
+  ], // depends on CURRENT_EVENT
+  [AppFeature.EVENTS_LIST]: AppFeature.EVENTS_LIST,
+};
+
+// ! Relevant only for map mode
+// WIP implementation
+// WatchList format {eventToWatch: null, ...}
+export function buildWatchList(hasFeature: (f: AppFeatureType) => boolean) {
+  const o = {};
+
+  forEach(APPEVENT_TO_FEATURE, (value, appEvent) => {
+    if (value === null || every(castArray(value), (f) => hasFeature(f))) {
```

Comment:
to not to castArray we can set APPEVENT_TO_FEATURE keys all as arrays
will it be better? 
```
const APPEVENT_TO_FEATURE = {
  [AppFeature.CURRENT_EVENT]: [AppFeature.CURRENT_EVENT], // can be error
  _done_layersGlobalResource: [AppFeature.LAYERS_IN_AREA], // list of layers
  [AppFeature.ANALYTICS_PANEL]: [AppFeature.ANALYTICS_PANEL, AppFeature.CURRENT_EVENT], // not in EPIG, depends on CURRENT_EVENT
  // _done_areaLayersDetailsResourceAtom: null, // can be disabled in url
  _done_layersInAreaAndEventLayerResource: [
    AppFeature.LAYERS_IN_AREA,
    AppFeature.CURRENT_EVENT,
  ], // depends on CURRENT_EVENT
  [AppFeature.EVENTS_LIST]: [AppFeature.EVENTS_LIST],
};
```